### PR TITLE
Add a Refresh Token function to the OpenID Connect provider that also returns the refreshed OpenID id_token

### PR DESCRIPTION
**Pull request by subhan-nadeem**
[The ID token is a fundamental part of the OpenID connect refresh token flow](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) (`id_token`). but is not part of the OAuth flow.

The existing `RefreshToken` function in the OpenID Connect provider leverages the OAuth library's refresh token mechanism, ignoring the optionally refreshed `id_token`. As a result, a new function needs to be exposed (rather than changing the existing function, for backwards compatibility purposes) that also returns the `id_token` in the OpenID refresh token flow API response. 

This PR creates and exposes such a flow with the function `RefreshTokenWithIDToken`. I've written this logic internally for our project at Tesla because we cannot use the `RefreshToken` endpoint due to it discarding the `id_token` response field, but figured it should be a part of the original `goth` library since this logic conforms to the Open ID connect spec.

### CODE REVIEW

File: providers/openidConnect/openidConnect.go [hunk 1]
```diff
@@ -197,6 +214,46 @@ func (p *Provider) RefreshToken(refreshToken string) (*oauth2.Token, error) {
...
+
+	resp, err := p.Client().Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Non-200 response from RefreshToken: %d, WWW-Authenticate=%s", resp.StatusCode, resp.Header.Get("WWW-Authenticate"))
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
```
Comment (bentranter) [inline providers/openidConnect/openidConnect.go:244]: The error from `ioutil.ReadAll` should be checked and returned,
```suggestion
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
	    return nil, err
	}
```

**Review by bentranter — CHANGES_REQUESTED** [review]
One tiny change, after that, LGTM!

File: providers/openidConnect/openidConnect.go (hunk 1 above)
Comment (subhan-nadeem) [reply providers/openidConnect/openidConnect.go:244]: Addressed! Thank you

**Comment by subhan-nadeem** [comment]
Change addressed @bentranter , thank you

**Review by bentranter — APPROVED** [review]
LGTM, thanks again @subhan-nadeem!

**Comment by mrkschan** [comment]
Is there any specific reason not to parse `expires_in` of the refresh token response as per https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokenResponse?

Without such field, caller of `RefreshTokenWithIDToken()` has to parse the JWT again just to obtain the expiry.

**Pull request merged**